concordium-std/src/test_infrastructure.rs

```diff
@@ -1588,6 +1648,36 @@ impl<State: Serial + DeserialWithState<TestStateApi>> TestHost<State> {
         *self.contract_balance.borrow_mut() = amount;
     }
 
+    pub fn set_self_address(&mut self, address: ContractAddress) {
+        if self.missing_contracts.contains(&address) {
+            fail!("The self_address is marked as a missing contract address.")
+        }
+        self.contract_address = Some(address);
+    }
+
+    pub fn setup_account_balance(
+        &mut self,
+        address: AccountAddress,
+        account_balance: AccountBalance,
+    ) {
+        self.query_account_balances.borrow_mut().insert(address, account_balance);
+    }
+
+    pub fn setup_contract_balance(&mut self, address: ContractAddress, balance: Amount) {
+        self.query_contract_balances.borrow_mut().insert(address, balance);
+
+        // Update the contract balance if the address matches.
+        if let Some(self_address) = self.contract_address {
+            if self_address == address {
+                *self.contract_balance.borrow_mut() = balance;
+            }
+        }
```

Comment:
I wouldn't expect this to change the self balance. Perhaps we should fail instead.

We could add a helper function for querying the balance, which takes in the optional self balance and address along with the contract query map? So we can have a single location for the logic regarding this special case?

And perhaps we should change the name to `setup_query_contract_balance` to distinguish it more from the `setup_self_balance` method.

Reply:
I have renamed the setup functions and added a check as you suggested here, as well a check when setting self_address